client/me/billing-history/upcoming-charges-table.jsx

```diff
@@ -1,62 +1,63 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal Dependencies
  */
-var	purchasesPaths = require( 'me/purchases/paths' ),
-	TransactionsTable = require( './transactions-table' );
+import purchasesPaths from 'me/purchases/paths';
+import TransactionsTable from './transactions-table';
+import { getSitesForUpcomingTransactions } from 'state/selectors';
+
+class UpcomingChargesTable extends Component {
+	static propTypes = {
+		// Computed props
+		sites: PropTypes.object.isRequired,
+	}
 
-module.exports = React.createClass( {
+	renderTransaction = ( transaction ) => {
+		const { translate } = this.props;
+		const site = this.props.sites[ Number( transaction.blog_id ) ];
 
-	displayName: 'UpcomingChargesTable',
+		if ( ! site ) {
+			return null;
```

Comment:
> Clearing browser state, then visiting http://calypso.localhost:3000/me/purchases/billing now causes each transaction to be missing the Manage Purchase link. 

Well, this regression is one of the many cases where a `QuerySites` in layout would resolve it. The problem is that the sites are actually never requested while in /purchases, because in this PR we're no longer relying on sites-list. Perhaps we can try landing #13094 or at least parts of it? I'm sure this would unblock many other similar cases. cc @mcsf.

> It looks like we want the full site object here just for a site.slug and transaction id. Can we update getSitesForUpcomingTransactions to just return ids, and use site id in place of the site slug? We could maybe then have the site purchase route redirect to the correct slug since we'd likely need to fetch a site in that case.

In theory, we could do that, but I feel it's going way outside of this PR's goal. There are currently 6 different routes that rely on the slug being there, and not an ID instead; and we'd have to add just another redirect. And would that really help? We'd still have to fetch a site before displaying the transaction details, which doesn't feel very reactive to me. I'd expect for the "Manage Purchase" links to be either hidden (or a placeholder be shown there) while the sites are loading, and then, when the site is loaded, the link would be shown. This could be resolved with what I've suggested above (landing #13094 or parts of it). Does that sound like a solution we could go with?